src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/DeflateManagedStream.cs

```diff
@@ -14,38 +14,29 @@ internal sealed partial class DeflateManagedStream : Stream
     {
         internal const int DefaultBufferSize = 8192;
 
-        private Stream _stream;
+        private Stream? _stream;
         private bool _leaveOpen;
-        private InflaterManaged _inflater;
-        private byte[] _buffer;
+        private InflaterManaged _inflater = null!;
+        private byte[] _buffer = null!;
 
         private int _asyncOperations;
 
         // A specific constructor to allow decompression of Deflate64
-        internal DeflateManagedStream(Stream stream, ZipArchiveEntry.CompressionMethodValues method, long uncompressedSize)
+        internal DeflateManagedStream(Stream stream, ZipArchiveEntry.CompressionMethodValues method, long uncompressedSize = -1)
         {
             if (stream == null)
                 throw new ArgumentNullException(nameof(stream));
             if (!stream.CanRead)
                 throw new ArgumentException(SR.NotSupported_UnreadableStream, nameof(stream));
-
-            InitializeInflater(stream, false, null, method, uncompressedSize);
-        }
-
-        /// <summary>
-        /// Sets up this DeflateManagedStream to be used for Inflation/Decompression
-        /// </summary>
-        internal void InitializeInflater(Stream stream, bool leaveOpen, IFileFormatReader reader = null, ZipArchiveEntry.CompressionMethodValues method = ZipArchiveEntry.CompressionMethodValues.Deflate, long uncompressedSize = -1)
-        {
-            Debug.Assert(stream != null);
-            Debug.Assert(method == ZipArchiveEntry.CompressionMethodValues.Deflate64);
             if (!stream.CanRead)
                 throw new ArgumentException(SR.NotSupported_UnreadableStream, nameof(stream));
 
-            _inflater = new InflaterManaged(reader, method == ZipArchiveEntry.CompressionMethodValues.Deflate64 ? true : false, uncompressedSize);
+            Debug.Assert(method == ZipArchiveEntry.CompressionMethodValues.Deflate64);
+
+            _inflater = new InflaterManaged(null, method == ZipArchiveEntry.CompressionMethodValues.Deflate64 ? true : false, uncompressedSize);
 
             _stream = stream;
-            _leaveOpen = leaveOpen;
+            _leaveOpen = false;
```

Comment:
Do we need to initialize _leaveOpen? Wouldn't it's default runtime value be `false`? If we need to initialize it, I would initialize it on the declaration and make it readonly, this is the only place where we initialize it.

Reply:
I just followed existing implementation, but yes as it will be false by default I could remove this row. I see its only initialized here which means its always be false and not really used in this class, so might remove this field instead. But seems it could be true in write mode somehow we are not using 64 bit deflator in write mode. Not sure if that is an issue @ericstj any opinion?

Reply:
Was need to remove as compiler complaining about unused variable